Minio/DataModel/ObjectOperationsArgs.cs

```diff
@@ -853,4 +855,357 @@ public override RestRequest BuildRequest(RestRequest request)
+                    ((this.LegalHoldEnabled == true)?"ON":"OFF"),
+                    ParameterType.HttpHeader);
+            }
+            return request;
+        }
+
+        public NewMultipartUploadArgs WithSSEHeaders(Dictionary<string, string> hdr)
+        {
+            this.WithHeaders(hdr);
+            return this;
+        }
+    }
+
+    public class PutObjectPartArgs : PutObjectArgs
+    {
+        public PutObjectPartArgs()
+        {
+            this.RequestMethod = Method.PUT;
+        }
+
+        public override void Validate()
+        {
+            base.Validate();
+            if (string.IsNullOrEmpty(this.UploadId) || string.IsNullOrWhiteSpace(this.UploadId))
+            {
+                throw new ArgumentNullException(nameof(UploadId) + " not assigned for PutObjectPart operation.");
+            }
+        }
+        public new PutObjectPartArgs WithBucket(string bkt)
+        {
+            return (PutObjectPartArgs)base.WithBucket(bkt);
+        }
+
+        public new PutObjectPartArgs WithObject(string obj)
+        {
+            return (PutObjectPartArgs)base.WithObject(obj);
+        }
+
+        public new PutObjectPartArgs WithObjectSize(long size)
+        {
+            return (PutObjectPartArgs)base.WithObjectSize(size);
+        }
+
+        public new PutObjectPartArgs WithHeaders(Dictionary<string, string> hdr)
+        {
+            return (PutObjectPartArgs)base.WithHeaders(hdr);
+        }
+
+        public new PutObjectPartArgs WithRequestBody(object data)
+        {
+            return (PutObjectPartArgs)base.WithRequestBody(data);
+        }
+        public new PutObjectPartArgs WithSSEHeaders(Dictionary<string, string> sseHeaders)
+        {
+            this.SSEHeaders = this.SSEHeaders ?? new Dictionary<string, string>();
+            if (sseHeaders != null)
+            {
+                this.SSEHeaders = this.SSEHeaders.Concat(sseHeaders).GroupBy(ele => ele.Key).ToDictionary(ele => ele.Key, ele => ele.First().Value);
+                if (this.SSE != null &&
+                (this.SSE.GetType().Equals(EncryptionType.SSE_S3) ||
+                    this.SSE.GetType().Equals(EncryptionType.SSE_KMS)))
+                {
+                    this.SSEHeaders.Remove(Constants.SSEGenericHeader);
+                    this.SSEHeaders.Remove(Constants.SSEKMSContext);
+                    this.SSEHeaders.Remove(Constants.SSEKMSKeyId);
+                }
+            }
+            return this;
+        }
+
+        public new PutObjectPartArgs WithStreamData(Stream data)
+        {
+            return (PutObjectPartArgs)base.WithStreamData(data);
+        }
+        public new PutObjectPartArgs WithContentType(string type)
+        {
+            return (PutObjectPartArgs)base.WithContentType(type);
+        }
+
+        public new PutObjectPartArgs WithUploadId(string id)
+        {
+            return (PutObjectPartArgs)base.WithUploadId(id);
+        }
```

Comment:
The SDK was masking lower level API's like multipart upload earlier and making the decision internally on performing a multipart upload vs single part. Is this being added to published API for a specific reason? If upload part is exposed in the API, then it ensues that NewMultipart and CompleteMultipart need to be added as well. I thought the purpose of this PR was just to standardize args in existing API calls

Reply:
PutObjectPartAsync, NewMultipartUploadAsync, CompleteMultipartUploadAsync are all private functions. If following with the previous implementation.
I'll make PutObjectPartArgs visibility internal.